### ISSUE TITLE
How to debug MooseX::Extended Modules?

### ISSUE THREAD

**Issue by thdrmz** [Ovid/moosex-extended#45]
```
./Build testdb --verbose=1 --test_files 02-greeter.t
---- 02-greeter.t ----
use Test2::V0 '!meta', -target=>q(Greeter);
---- result ----
t/02-greeter.t .. 
Loading DB routines from perl5db.pl version 1.60
Editor support available.

Enter h or 'h h' for help, or 'man perldebug' for more help.

TODO: Unsure how to find argcheck op within dbstate at /home/..../Greeter/blib/lib/Greeter.pm line 82.
 at /home/....//Greeter/blib/lib/Greeter.pm line 82.
        require Greeter.pm called at /usr/share/perl5/Test2/Tools/Target.pm line 41
        Test2::Tools::Target::import_into("Test2::Tools::Target", "main", "Greeter") called at /usr/share/perl5/Test2/V0.pm line 125
        Test2::V0::import("Test2::V0", "!meta", "-target", "Greeter") called at t/02-greeter.t line 1
        main::BEGIN() called at /home/....//Greeter/blib/lib/Greeter.pm line 82
        eval {...} called at /home/....//Greeter/blib/lib/Greeter.pm line 82
Compilation failed in require at /usr/share/perl5/Test2/Tools/Target.pm line 41.
 at /usr/share/perl5/Test2/Tools/Target.pm line 41.
        Test2::Tools::Target::import_into("Test2::Tools::Target", "main", "Greeter") called at /usr/share/perl5/Test2/V0.pm line 125
        Test2::V0::import("Test2::V0", "!meta", "-target", "Greeter") called at t/02-greeter.t line 1
        main::BEGIN() called at t/02-greeter.t line 1
        eval {...} called at t/02-greeter.t line 1
BEGIN failed--compilation aborted at t/02-greeter.t line 1.
 at t/02-greeter.t line 1.
Debugged program terminated.  Use q to quit or R to restart,
```

Greeter.pm:

```
package Greeter;
our $VERSION = '0.01';
use 5.26.0;
use strict;
use warnings;
use List::Util qw(first);
use MooseX::Extended
	types=>[qw(Str ArrayRef)],
	includes=>[qw(multi)],
	excludes=>[qw(autoclean carp)]; 

param guests => (isa => 'ArrayRef[Str]');
param members => (isa=>'ArrayRef[Greeter::Member]');

param unknown_greeting => (
	is => 'ro',
	isa => 'Str',
	required=>0,
	default=>q(Hello __NAME__, I don't know you, do I?));
#
param guest_greeting => (
	is => 'ro',
	isa => 'Str',
	required=>0,
	default=>q(Hello __NAME__, I hope you are having a nice visit));
#
sub _greet_person($o,$name) {
	my $gr;
	my($mem, $guest);
	if( $mem = first { $_->name eq $name } @{$o->members} ) {
		$gr = $mem->greeting;
	} elsif ( $guest = first { $_ eq $name } @{$o->guests} ) {
		$gr = $o->_greet_guest($name);
	} else {
		$gr = $o->_greet_unknown($name);
	}
	return($gr);
} 
#
sub _greet_unknown($o,$name) {
	my $gr=$o->unknown_greeting;
	$gr=~s/__NAME__/$name/;
	return($gr);
}
#
sub _greet_guest($o,$n) {
	my $gr=$o->guest_greeting;
	$gr=~s/__NAME__/$n/;
	return($gr);
}
#
sub _greet_other_person($o,$gr,$gt) {
	my $greeting;
	my $members_greeting = first { $_->name eq $gr } @{$o->members};
	my $member_to_greet = first { $_->name eq $gt } @{$o->members};
	my $guests_greeting = first { $_ eq $gr } @{$o->guests};
	my $guest_to_greet = first { $_ eq $gt } @{$o->guests};
	if ( $members_greeting && $member_to_greet ) {
		$gr = $members_greeting->greet_guest($member_to_greet->name);
	} elsif ($members_greeting && $guest_to_greet ) {
		$gr = $members_greeting->greet_guest($guest_to_greet);
	} elsif ($guests_greeting && $guest_to_greet) {
		$gr = $o->_greet_guest($gt);
	} elsif ($member_to_greet) {
		$gr =$member_to_greet->greeting;
	} elsif( !($members_greeting || $guests_greeting) && $guest_to_greet) {
		$gr = $o->_greet_guest($guest_to_greet);
	} else {
		$gr = $o->_greet_unknown($gt);
	}
	return($gr);
}
#
multi sub greet($o,$gn,$gt) {
	my $gr;
	if(!defined($gt) || $gn eq $gt) {
		$gr = $o->_greet_person($gn);
	} else {
		$gr= $o->_greet_other_person($gn, $gt);
	}
	return($gr);
}
multi sub greet($o,$gn) {
	return($o->_greet_person($gn));
}
0;
```

**Comment by Ovid** [comment]
Hi @thdrmz : I've been on holiday for the past week. I'll try to take a look at this in the next few days.

**Comment by Ovid** [comment]
This appears to be a bug in Syntax::Keyword::MultiSub: https://rt.cpan.org/Ticket/Display.html?id=143997

Note that there appears to also be an issue with RT because it’s not showing up on the issues list for that module. I've alerted the RT cabal and also emailed the module author directly.

**Comment by Ovid** [comment]
The follow patch to `Syntax::Keyword::MultiSub` fixes the issue:

```
--- old/lib/Syntax/Keyword/MultiSub.xs  2021-12-16 10:59:30 +0000
+++ new/lib/Syntax/Keyword/MultiSub.xs  2022-08-12 10:23:06 +0000
@@ -129,6 +129,7 @@
 redo:
     switch(o->op_type) {
       case OP_NEXTSTATE:
+      case OP_DBSTATE:
         o = o->op_next;
         goto redo;
```

Paul will include this in his next release. I'll leave this issue open until that time.

**Comment by Ovid** [comment]
This is now resolved with the 0.35 release.
